components/org.wso2.carbon.identity.branding.preference.management.core/src/test/java/org/wso2/carbon/identity/branding/preference/management/core/BrandingPreferenceManagerImplTest.java

```diff
@@ -0,0 +1,342 @@
+    private BrandingPreferenceManagerImpl brandingPreferenceManagerImpl;
+
+    @BeforeMethod
+    public void setUp() throws Exception {
+
+        initMocks(this);
+        setCarbonHome();
+        setCarbonContextForTenant(SUPER_TENANT_DOMAIN_NAME, SUPER_TENANT_ID);
+        brandingPreferenceManagerImpl = new BrandingPreferenceManagerImpl();
+
+        configurationManager = ConfigurationManagementUtils.getConfigurationManager();
+        BrandingPreferenceManagerComponentDataHolder.getInstance().setConfigurationManager(configurationManager);
+
+        BrandingPreferenceManagerComponentDataHolder.getInstance().setIdentityEventService(identityEventService);
+        doNothing().when(identityEventService).handleEvent(any(Event.class));
+    }
+
+    @DataProvider(name = "brandingPreferenceDataProvider")
+    public Object[][] brandingPreferenceDataProvider() throws Exception {
+
+        BrandingPreference brandingPreference1 = new BrandingPreference();
+        brandingPreference1.setType(ORGANIZATION_TYPE);
+        brandingPreference1.setName(SUPER_TENANT_DOMAIN_NAME);
+        brandingPreference1.setLocale(DEFAULT_LOCALE);
+        brandingPreference1.setPreference(getPreferenceFromFile("sample-preference-1.json"));
+
+        BrandingPreference brandingPreference2 = new BrandingPreference();
+        brandingPreference2.setType(ORGANIZATION_TYPE);
+        brandingPreference2.setName(SAMPLE_TENANT_DOMAIN_NAME_ABC);
+        brandingPreference2.setLocale(DEFAULT_LOCALE);
+        brandingPreference2.setPreference(getPreferenceFromFile("sample-preference-2.json"));
+
+        return new Object[][]{
+                {brandingPreference1, SUPER_TENANT_DOMAIN_NAME, SUPER_TENANT_ID},
+                {brandingPreference2, SAMPLE_TENANT_DOMAIN_NAME_ABC, SAMPLE_TENANT_ID_ABC},
+        };
+    }
+
+    @Test(dataProvider = "brandingPreferenceDataProvider")
+    public void testAddBrandingPreference(Object brandingPreference, String tenantDomain, int tenantId)
+            throws Exception {
+
+        setCarbonContextForTenant(tenantDomain, tenantId);
+        BrandingPreference inputBP = (BrandingPreference) brandingPreference;
+
+        // Adding new branding preference.
+        BrandingPreference addedBP = brandingPreferenceManagerImpl.addBrandingPreference(inputBP);
+        Assert.assertEquals(addedBP.getPreference(), inputBP.getPreference());
+        Assert.assertEquals(addedBP.getName(), inputBP.getName());
+
+        //  Retrieving added branding preference.
+        BrandingPreference retrievedBP = brandingPreferenceManagerImpl.getBrandingPreference
+                (inputBP.getType(), inputBP.getName(), inputBP.getLocale());
+        Assert.assertEquals(retrievedBP.getPreference(), inputBP.getPreference());
+        Assert.assertEquals(retrievedBP.getName(), inputBP.getName());
+
+        // Deleting added branding preference.
+        brandingPreferenceManagerImpl.deleteBrandingPreference(inputBP.getType(), inputBP.getName(),
```

Comment:
if one of the assertions above fails, this line will not execute. If the other tests also use same params for branding preference other tests may also fail.

Reply:
Yes it's expected. We can't extract this delete method to a `AfterMethod` since we need branding preference name, type and locale to delete it.